### DISCUSSION TITLE
Add option for using leader key is global shortcut

### DISCUSSION THREAD

**Discussion by Mishkun** [JetBrains/ideavim#345] [Ideas]
I really enjoy spacemacs-esque config where every non-editing command (e.g. refactor/split/run) starts with <SPACE> in normal mode. However, leader key (and everything ideaVim related) works only when focusing editor. If focusing other tool window, these magic leader combinations do not work. It would be good to add an option to start leader sequences from any window.

Looks like It could be implemented by adding some shortcut on vim plugin startup and verifying for active speed searches. I started to implement this by myself, but I struggle with two questions:
1) How do I get root `JComponent` to register shortcut globally with `EventFacade.getInstance().registerCustomShortcutSet`?
2) How do I call Vim mapping after that? Seems like `VimPlugin.getKey().getKeyMapping` gets only keystrokes list, but do not interprets them

**Comment by Mishkun** [comment]
I wrote a separate plugin for this https://github.com/Mishkun/ataman-intellij

**Reply by AlexPl292** [reply]
Hey, congratulations with a great plugin, I see a lot of great responses about it!
I apologise, I completely missed your question from my view. As I understand, there is not more need to answer them.
We have future plans to teach IdeaVim to work everywhere and not just in the editor, but this is not an immediate priority at the moment.

**Reply by arabshapt** [reply]
@AlexPl292 , is there no way to do the same what glimpse plugin for vscode does in intellij? Ataman-intellij is very similar to it, but you can't type the sequence fast. If you do then some of the keys do not get recognised. How would you trigger leader from anywhere in the IDE in a way that lets you be fast and not worry about your speed?

**Reply by Mishkun** [reply]
@arabshapt I think that ataman's perfomance can be improved significantly, but to do that I actually need to understand how swing works

**Reply by arabshapt** [reply]
@Mishkun The popup and menu is good but for me the ability to do a fast sequence of taps is the most important feature. Because the help is only needed at the beginning of learning phase and after that you just type without looking. That is the idea of whichkey(to only show after a delay(meaning that you are looking for an action but don't know the sequence yet.))
I wish it were possible to just record the key sequence and then go in your config and execute the action...
So it is not just about performance of UI, but that it is connected to the speed of UI at all. It should be decoupled from it to make it possible to be very fast(even when you PC is slow)

**Reply by arabshapt** [reply]
Vspacecode plugin for vs code does a pretty good job, but I need something similar in intellij. Calling functions from anywhere in the ide
